### DISCUSSION TITLE
Include a Cargo.lock

### DISCUSSION THREAD

**Discussion by arduano** [SeaQL/FireDBG.for.Rust#21] [Bug]
Hi! I was just attempting to make a Nix package based on this repository, and doing that requires having a Cargo.lock file in the repository (or manually providing one myself, which I'll do for now). It's generally standard practice to include a Cargo.lock, though correct me if the unique requirements of this project mean that one can't be included.

**Comment by tyt2y3** [comment]
You're right, I think we simply git ignored our Cargo.lock. We'd include one from the next release onwards.

**Reply by tyt2y3** [reply]
Now includeded 4240720b98486c050e6441e3f2f414b4ffa36dc7

**Reply by arduano** [reply]
@tyt2y3 Hey, just FYI but generally there's supposed to be only 1 Cargo.lock per workspace. Not that it's a significant issue, but just wondering, is this caused by your build process? I noticed that in the build scripts you were using `--manifest-path` instead of `--project`, the latter being the more standard option to use. E.g. `cargo build --project=indexer`.

**Reply by tyt2y3** [reply]
You are right, I messed things up, check the Cargo.lock at the root 13785e4a92460bd755229542926d757176b30f7f

**Reply by arduano** [reply]
Perfect, probably just needs updating the building.md file. I was slightly wrong in my above comment, where it's not `cargo build --project=indexer`, instead it's `cargo build --project=firedbg-stream-indexer` or any other package name (instead of folder name). Or `cargo build --all` also should probably work. Though I'm also slightly confused why `--release` isn't included there.

**Reply by arduano** [reply]
Also, annoyingly, I got it building fine on NixOS, but the extension still wouldn't detect the installed binaries. If you don't mind, could you please point me to the extension code that checks if the binaries exist? I couldn't find that code for some reason.

**Reply by tyt2y3** [reply]
The vscode extension is currently not open source, but there is an option to set the path for the binaries
![image](https://github.com/SeaQL/FireDBG.for.Rust/assets/1782664/44a1de90-950b-4cf5-9013-1b0012b82f8c)
If unset, it defaults to `~/.cargo/bin`

**Reply by arduano** [reply]
Ah I see, will keep that in mind, thanks!
